src/book.js

```diff
@@ -446,29 +352,145 @@ export class Book {
     } else if (this.format == '8_zine') {
       previewPdf = await this.buildSheets(this.filename, this.book.page_8_zine_builder());
     }
-    console.log('Attempting to generate preview for ', previewPdf);
 
-    if (previewPdf != null) {
-      const pdfDataUri = await previewPdf.saveAsBase64({ dataUri: true });
-      const viewerPrefs = previewPdf.catalog.getOrCreateViewerPreferences();
-      viewerPrefs.setHideToolbar(false);
-      viewerPrefs.setHideMenubar(false);
-      viewerPrefs.setHideWindowUI(false);
-      viewerPrefs.setFitWindow(true);
-      viewerPrefs.setCenterWindow(true);
-      viewerPrefs.setDisplayDocTitle(true);
-
-      previewFrame.src = pdfDataUri;
-      previewFrame.style.width = `450px`;
-      const height = (this.papersize[1] / this.papersize[0]) * 500;
-      previewFrame.style.height = `${height}px`;
-      previewFrame.style.display = '';
-    }
+    if (previewPdf != null) await this.displayPreview(previewPdf);
 
     if (!isPreview) return this.saveZip();
     else return Promise.resolve(1);
   }
 
+  /**
+   * Generates the signatures for a file from the input
+   * @param {boolean} isPreview - whether we're generating files for a preview, or to save
+   * @param {Object[]} signatures - object to organize the generated signatures on
+   * @returns reference to a PDF for preview, or null, depending on isPreview
+   */
+
+  async generateClassicFiles(isPreview, signatures) {
+    // Only generate the first signature for preview
+    const pagesArr = isPreview ? this.rearrangedpages.slice(0, 1) : this.rearrangedpages;
+    let previewPdf = null;
+    const makeSignatures = async () => {
+      const tasks = pagesArr.map(async (pages, i) => {
+        console.log(pages);
+        signatures[i] = { name: `${this.filename}_signature${i}` };
+        [signatures[i].front, signatures[i].back] = await this.createSignatures({
```

Comment:
As @acestronautical pointed out -- the scope here changed. 
I'm squinting and not fresh on JS scope (but can be! used to be!) so I can't just say what the answer is -- let me know if you want help fixing it

Reply:
err-- I'd argue that the typo was the other way around -- I believe that function creates a single signature at a time -- believe I actually ran into quite a bit of pain with that before I figured it out, the plural throwing me off and giving the wrong impression. 

I tried to audit it locally but it appears there's still a bug in producing PDFs -- are you able to do so off the main branch?

I got a different, hard to read error w/ 
https://momijizukamori.github.io/bookbinder-js/pr-preview/pr-116/?printFile=aggregated&rotatePage=true&paperSize=LETTER&pageLayout=quarto&pdfEdgeMarks=true&sigOrderMarks=true&pagePositioning=binding_aligned&mainForeEdgePaddingPt=40&sigLength=2&customSigLength=0&wackySpacing=wacky_gap&flyleafs=0&sewingMarksEnabled=true&sewingMarkLocation=only_in&sewingMarksMarginPt=50&sewingMarksAmount=2&sewingMarksTapeWidthPt=40

Reply:
Poking that, it seems to be only if the signature order marks are enabled - the others work fine. I'll see what I can do with that (and change that pluralization, heh)